### PR TITLE
Add bors

### PR DESCRIPTION
**Pull request by Hoverbear**
## What have you changed? (mandatory)

Enables @bors for TiKV.

Learn about [bors](http://bors.tech/), or see the `CONTRIBUTING.md` changes in this PR.

**What is Bors?:** Bors is a bot who will help us merge PRs faster and reduce CI wait times!
**How?:** Once a PR has two approvals you will be able to do `bors r+` as a maintainer of TiKV to have bors go ahead and try to merge the PR. Bors will automatically merge if there are no conflicts, and if a conflict breaks the build it will  report it. For small changes, bors supports a `bors rollup` command to have it try multiple PRs at once, we have not tried this though. Bors also supports setting priorities on the queue, so important builds can be prioritized.
**Why?:**  Right now our team waits a long time for CI builds, and if another PR is merged all others must rebuild. This is time consuming and slow. Bors will do this for us automagically in a queue.
**When?:** I will need some time to figure out how to work it with CircleCI and Jenkins. Right now we have it working for AppVeyor and Travis-CI. If you'd like, I am very happy to set up bors on any libraries you are working on as well, but a large project gets the most benefit.

This introduces bors in an optional, supplementary manner.

PRs do not **yet** require bors to be used to be merged. This is intended for the future work, **that PR will impact your workflow!!!**

This is a **best effort** PR, as we may need to tinker with the CI to get bors properly detecting the statuses of CircleCI and (later) Jenkins. Expect follow ups.

## What are the type of the changes? (mandatory)

New tooling.

## How has this PR been tested? (mandatory)

It adds CI infrastructure, so cannot be practically tested. However if this fails it will not impact our contributors or workflow.

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

Only contributing docs, only for reviewers.

## Does this PR affect tidb-ansible update? (mandatory)

Nope

## Refer to a related PR or issue link (optional)

https://github.com/pingcap/raft-rs/pull/137
https://github.com/pingcap/raft-rs/pull/134

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,6 +23,24 @@ This is a rough outline of what a contributor's workflow looks like:
...
+- Review it. Give complete feedback, make sure the contributor can understand your suggestions. If needed, help them.
+- If you approve it, use the 'Submit Review' button and choose 'Approve.'
+- If you are the first approver, request someone else to review it as well.
+- If you are the second approver, you can leave `bors r+` in the review or as a comment.
+    + If the change is trivial, or you are sure it cannot fail, you can use `bors rollup` to allow bors to batch a merge.
+- bors will merge the code after a time so long as there are no merge conflicts with master, and the tests pass.
+    + **You do not need to hit the 'Update Branch' button unless there are conflicts or you think it will fail.**
+- You will recieve a notification for the PR saying it was merged, or it failed.
+
+If you need support using bors you should consult [the guide](https://bors.tech/documentation/getting-started/), you may also ping @hoverbear.
```
Comment (ice1000) [inline CONTRIBUTING.md]: Is this `@` gonna work as a link?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -68,4 +86,4 @@ you can use one of some generic reasons like "Improve documentation.",
 
 The project now enables [DCO check](https://github.com/probot/dco#how-it-works) and the commit message must contain a `Signed-off-by` line for [Developer Certificate of Origin](https://developercertificate.org/)
 
-You can use option `-s` for `git commit` to automatically add a `Signed-off-by` to the commit message. 
\ No newline at end of file
+You can use option `-s` for `git commit` to automatically add a `Signed-off-by` to the commit message.
```
Comment (ice1000) [inline CONTRIBUTING.md]: Why is this line marked as a change by GitHub.. Did you do anything to it?

Comment (Hoverbear) [reply CONTRIBUTING.md]: It's the last line of a file, and typically this has a blank newline after it (eg `blah blah\n` is the raw content of a 1 line file). My Git config is is set to automatically add this to any file without it.

Notably, this is part of the POISIX standard: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206, so there isn't a harm to following this.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Hoverbear) [reply CONTRIBUTING.md]: I'm not entirely sure, I'll make it an explicit link.

**Comment by Hoverbear** [comment]
This is blocked by #3623.

**Comment by Hoverbear** [comment]
@BusyJay noted that bors currently doesn't squash PRs on merge. :(

**Comment by Hoverbear** [comment]
@siddontang If we must have squashed merges then bors no longer provides us any benefit (since then they would need to squash their commits after every small feedback, it would make reviewing very hard).

If we can accept non-squashed merges then we can totally use bors and it can still help us. In this case we can encourage our developers to avoid having lots and lots of `change` or `small fix` commits before the initial review. So it might encourage us to use better commit style.

What do you think?

**Pull request closed without merging**